api/src/test/java/org/openmrs/util/DatabaseIT.java

```diff
@@ -55,40 +57,62 @@ public Liquibase getLiquibase(String filename) throws LiquibaseException, SQLExc
 		Database liquibaseConnection = DatabaseFactory.getInstance()
 		        .findCorrectDatabaseImplementation(new JdbcConnection(getConnection()));
 		
-		liquibaseConnection.setDatabaseChangeLogTableName("LIQUIBASECHANGELOG");
-		liquibaseConnection.setDatabaseChangeLogLockTableName("LIQUIBASECHANGELOGLOCK");
+		liquibaseConnection.setDatabaseChangeLogTableName("liquibasechangelog");
+		liquibaseConnection.setDatabaseChangeLogLockTableName("liquibasechangeloglock");
 		
 		return new Liquibase(filename, new ClassLoaderResourceAccessor(getClass().getClassLoader()), liquibaseConnection);
 	}
 	
-	protected void initializeDatabase() throws SQLException, ClassNotFoundException {
-		String driver = "org.h2.Driver";
-		Class.forName(driver);
+	protected void initializeDatabase() throws ClassNotFoundException {
+		if(!useInMemoryDatabase()) {
+			setupContainerDB();
+		} else {
+			System.setProperty("databaseDialect", H2LessStrictDialect.class.getName());
+		}
 	}
-	
+
+	private void setupContainerDB() throws ClassNotFoundException {
+		Containers.ensureDatabaseRunning();
+		CONNECTION_URL = System.getProperty("databaseUrl");
+	}
+
 	protected void updateDatabase(String filename) throws Exception {
 		Liquibase liquibase = getLiquibase(filename);
 		liquibase.update(new Contexts(CONTEXT));
 		liquibase.getDatabase().getConnection().commit();
 	}
-	
+
 	protected void dropAllDatabaseObjects() throws SQLException {
-		Connection connection = getConnection();
-		Statement statement = null;
-		try {
-			statement = connection.createStatement();
-			String query = "DROP ALL OBJECTS";
-			statement.execute(query);
-		}
-		catch (JdbcSQLNonTransientException e) {
+		try (Connection connection = getConnection()) {
+			Statement statement = connection.createStatement();
+			if (useInMemoryDatabase()) {
+				String query = "DROP ALL OBJECTS";
+				statement.execute(query);
+			} else {
+				if ("postgres".equalsIgnoreCase(System.getProperty("database"))) {
+					connection.setAutoCommit(true);
+					String dropTables =
+						"DO $$ DECLARE " +
+							"    r RECORD; " +
+							"BEGIN " +
+							"    FOR r IN (SELECT tablename FROM pg_tables WHERE schemaname = 'public') " +
+							"    LOOP " +
+							"        EXECUTE 'DROP TABLE IF EXISTS ' || quote_ident(r.tablename) || ' CASCADE'; " +
+							"    END LOOP; " +
+							"END $$;";
+					statement.execute(dropTables);
+					connection.setAutoCommit(false);
+				} else if ("mysql".equalsIgnoreCase(System.getProperty("database"))) {
+					statement.execute("DROP DATABASE " + System.getProperty("databaseName") + ";");
+					statement.execute("CREATE DATABASE " + System.getProperty("databaseName") + ";");
```

Comment:
Could also explain a bit how creating a database is related to dropping all database objects?

Reply:
I think it's more convenient to simply drop and recreate the database in MySQL for this.

Reply:
Did you test these and confirm that they work for mysql and postgres?

Reply:
Yes

Reply:
Why don't we just do this for both PG and MySQL?

Reply:
postgres does not support dropping a database we are already connected to.